test/controllers/GuidanceControllerSpec.scala

```diff
@@ -25,162 +25,206 @@ import play.api.test.FakeRequest
 import play.api.test.Helpers._
 import uk.gov.hmrc.play.bootstrap.tools.Stubs.stubMessagesControllerComponents
 import uk.gov.hmrc.http.SessionKeys
+import forms.NextPageFormProvider
+import models.ui._
+import play.api.test.CSRFTokenHelper._
 
 import scala.concurrent.Future
 
 class GuidanceControllerSpec extends BaseSpec with GuiceOneAppPerSuite {
 
-  "Calling the scratch process endpoint with a valid UUID" should {
+  trait QuestionTest extends MockGuidanceService {
+    val path = "/some-path"
+    val relativePath = path.drop(1)
+    val processId = "ext90002"
+    val fakeRequest = FakeRequest("GET", path).withSession(SessionKeys.sessionId -> processId).withFormUrlEncodedBody().withCSRFToken
+
+    val ans1 = Answer(Text("ANS1", "ANS1"), Some(Text("","")), "/hello")
+    val ans2 = Answer(Text("ANS2", "ANS2"), Some(Text("","")), "/world")
+    val expectedPage: Page = QuestionPage(
+      path,
+      Question(Text("QUESTION","QUESTION"), Seq(Paragraph(Text("QUESTION","QUESTION"))), Seq(ans1, ans2))
+    )
+
+    MockGuidanceService
+      .getPage(path, processId)
+      .returns(Future.successful(Some(expectedPage)))
+
+    val errorHandler = app.injector.instanceOf[config.ErrorHandler]
+    val view = app.injector.instanceOf[views.html.standard_page]
+    val questionView = app.injector.instanceOf[views.html.question_page]
+    val target = new GuidanceController(errorHandler, 
+                                             view, 
+                                             questionView, 
+                                             new NextPageFormProvider(), 
+                                             mockGuidanceService, 
+                                             stubMessagesControllerComponents())
+  }
+
+
+  "Calling a valid URL path to a Question page in a process" should {
+
+    "return an OK response" in new QuestionTest {
+      val result: Future[Result] = target.getPage(relativePath)(fakeRequest)
+      status(result) mustBe Status.OK
+    }
+
+    "be a HTML response" in new QuestionTest {
+      val result: Future[Result] = target.getPage(relativePath)(fakeRequest)
+      contentType(result) mustBe Some("text/html")
+    }    
+
+  }
+
+  "Submitting a blank Question page form" should {
+
+    "return a BadRequest response" in new QuestionTest {
+      val result: Future[Result] = target.submitPage(relativePath)(fakeRequest)
+      status(result) mustBe Status.BAD_REQUEST
+    }
+  }
+
+  "Submitting an answered Question page form" should {
```

Comment:
There is no test for a submission with form errors. Would this be difficult to simulate?

Reply:
will add

Reply:
Fixed